sys/net/vaina/vaina.c

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (C) 2020 Locha Inc
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net_vaina
+ * @{
+ *
+ * @file
+ * @brief       VAINA - Versatile Address Interface | Network Administration
+ *
+ * @author      Jean Pierre Dudey <jeandudey@hotmail.com>
+ * @}
+ */
+
+#include "net/vaina.h"
+
+#include "net/sock/udp.h"
+
+#if IS_USED(MODULE_AODVV2)
+#include "net/aodvv2/client.h"
+#endif
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+#if ENABLE_DEBUG == 1
+static char _stack[THREAD_STACKSIZE_DEAFULT + THREAD_EXTRA_STACKSIZE_PRINTF];
+#else
+static char _stack[THREAD_STACKSIZE_DEFAULT];
+#endif
+
+/**
+ * @brief   UDP socket
+ */
+static sock_udp_t _sock;
+
+static int _parse_msg(vaina_msg_t *vaina, uint8_t *buf, size_t len)
+{
+    memset(vaina, 0, sizeof(vaina_msg_t));
+
+    if (len < 2) {
+        DEBUG_PUTS("vaina: invalid message size!");
+        return -EINVAL;
+    }
+
+    uint8_t type = buf[0];
+    uint8_t seqno = buf[1];
+
+    switch (type) {
+#if IS_USED(MODULE_AODVV2)
+        case VAINA_MSG_RCS_ADD:
+        case VAINA_MSG_RCS_DEL:
+            if (len < (2 + sizeof(ipv6_addr_t))) {
+                return -EINVAL;
+            }
+            vaina->msg = type;
+            vaina->seqno = seqno;
+            if (vaina->msg == VAINA_MSG_RCS_ADD) {
+                memcpy(&vaina->payload.add.ip, &buf[1], sizeof(ipv6_addr_t));
+            }
+            else {
+                memcpy(&vaina->payload.del.ip, &buf[1], sizeof(ipv6_addr_t));
+            }
+            break;
+#endif
+
+        default:
+            DEBUG_PUTS("vaina: invalid message type");
+            return -EINVAL;
+    }
+
+    return 0;
+}
+
+static int _process_msg(vaina_msg_t *msg)
+{
+    switch (msg->msg) {
+#if IS_USED(MODULE_AODVV2)
+        case VAINA_MSG_RCS_ADD:
+            if (aodvv2_client_add(&msg->payload.add.ip, 128, 1) == NULL) {
+                DEBUG_PUTS("vaina: client set is full");
+                return -ENOSPC;
+            }
+            break;
+
+        case VAINA_MSG_RCS_DEL:
+            aodvv2_client_delete(&msg->payload.del.ip);
+            break;
+#endif
+
+        default:
```

Comment:
default case should raise an event error or something to return

Reply:
Done.